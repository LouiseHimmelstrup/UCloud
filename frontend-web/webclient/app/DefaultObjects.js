import {Cloud} from "../authentication/SDUCloudObject"
import {tusConfig} from "./Configurations";
import Uppy from "uppy";

export const DefaultStatus = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended.",
};

export const RightsMap = {
    NONE: 0,
    READ: 1,
    READ_WRITE: 2,
    OWN: 3,
};

export const AnalysesStatusMap = {
    "PENDING": 0,
    "IN PROGRESS": 1,
    "COMPLETED": 2,
};

export const RightsNameMap = {
    NONE: "None",
    READ: "Read",
    READ_WRITE: "Read/Write",
    OWN: "Own",
};

export const SensitivityLevel = {
    "OPEN_ACCESS": "Open Access",
    "CONFIDENTIAL": "Confidential",
    "SENSITIVE": "Sensitive",
};

export const SensitivityLevelMap = {
    "OPEN_ACCESS": 0,
    "CONFIDENTIAL": 1,
    "SENSITIVE": 2,
};

const initializeUppy = (restrictions) => 
    Uppy.Core({
        autoProceed: false,
        debug: false,
        restrictions: restrictions,
        meta: {
            sensitive: false,
        },
        onBeforeUpload: () => {
            return Cloud.receiveAccessTokenOrRefreshIt().then((data) => {
                tusConfig.headers["Authorization"] = `Bearer ${data}`;
            });
        }
    }).use(Uppy.Tus, tusConfig);

export const initObject = {
    files: {
        files: [],
        filesShown: 10,
        filesPerPage: 10,
        currentFilesPage: 0,
        loading: false,
        path: "",
        filesInfoPath: "",
        filesInfoFile: null,
        projects: []
    },
    uppy: {
        uppyFiles: initializeUppy({maxNumberOfFiles: false}),
        uppyFilesOpen: false,
        uppyRunApp: initializeUppy({ maxNumberOfFiles: 1}),
        uppyRunAppOpen: false
    },
    status: {
        status: DefaultStatus,
        title: ""
    }
};