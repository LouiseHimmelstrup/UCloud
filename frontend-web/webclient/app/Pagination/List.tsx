import * as React from "react";
import { Grid, Message, Header } from "semantic-ui-react";
import { Page } from "Types";
import { DefaultLoading } from "LoadingIcon/LoadingIcon";
import * as Self from ".";
import "./pagination.scss";
import { ifPresent } from "UtilityFunctions";
import { RefreshButton } from "UtilityComponents";

interface ListProps {
    pageRenderer: (page: Page<any>) => React.ReactNode

    // List state
    loading?: boolean

    // Page results
    page: Page<any>

    // Error properties  
    errorMessage?: string | (() => React.ReactNode | null)
    customEmptyPage?: React.ReactNode

    // Callbacks
    onItemsPerPageChanged: (itemsPerPage: number) => void
    onPageChanged: (newPage: number) => void
    onRefresh?: () => void
    onErrorDismiss?: () => void
}

export class List extends React.PureComponent<ListProps> {
    constructor(props: ListProps) {
        super(props);
    }

    render() {
        const props = this.props;
        const body = this.renderBody();

        let errorComponent: React.ReactNode = null;
        if (typeof props.errorMessage == "string") {
            errorComponent = <Message color="red" onDismiss={props.onErrorDismiss}>{props.errorMessage}</Message>;
        } else if (typeof props.errorMessage == "function") {
            errorComponent = props.errorMessage();
        }

        const refreshButton = !!this.props.onRefresh ? (
            <RefreshButton
                className="pagination-float-right"
                loading={this.props.loading}
                onClick={this.props.onRefresh} 
            />
        ) : null;

        return (
            <React.Fragment>
                {errorComponent}
                {refreshButton}
                <Self.EntriesPerPageSelector
                    content="Items per page"
                    className="items-per-page-padding pagination-float-right"
                    entriesPerPage={props.page.itemsPerPage}
                    onChange={(perPage) => ifPresent(props.onItemsPerPageChanged, (c) => c(perPage))}
                />
                {body}
                <div>
                    <Self.Buttons
                        as="span"
                        currentPage={props.page.pageNumber}
                        toPage={(page) => ifPresent(props.onPageChanged, (c) => c(page))}
                        totalPages={Math.ceil(props.page.itemsInTotal / props.page.itemsPerPage)}
                    />
                </div>
            </React.Fragment>
        );
    }


    private renderBody(): React.ReactNode {
        const props = this.props;
        if (props.loading) {
            return <Grid centered verticalAlign="middle" columns={1}>
                <div className="pagination-loader">
                    <DefaultLoading loading size={undefined} className="pagination-list-loading" />
                </div>
            </Grid>
        } else {
            if (props.page == null || props.page.items.length == 0) {
                if (!props.customEmptyPage) {
                    return <div>
                        <Header as="h2">
                            No results.
                            <a
                                href="#"
                                onClick={(e) => { e.preventDefault(); ifPresent(props.onRefresh, (c) => c()) }}
                            >
                                {" Try again?"}
                            </a>
                        </Header>
                    </div>;
                } else {
                    return props.customEmptyPage
                }
            } else {
                return props.pageRenderer(props.page);
            }
        }
    }
}
