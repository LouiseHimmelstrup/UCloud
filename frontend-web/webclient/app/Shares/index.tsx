export { List } from "./List";
import { Store } from "redux";
import { AccessRightValues } from "Types";
import PromiseKeeper from "PromiseKeeper";

export interface ListState {
    promises: PromiseKeeper
    shares: SharesByPath[]
    errorMessage?: string,
    page: number,
    itemsPerPage: number
    loading: boolean
}

export interface ListProps {
    keepTitle?: boolean
    byPath?: string
}

export interface ListContext {
    store: Store
}

export interface Share {
    id: ShareId,
    sharedWith: String,
    rights: AccessRightValues[],
    state: ShareStateValues
}

// FIXME Singular instead of plural?

export type ShareStateValues = keyof typeof ShareState
export enum ShareState {
    REQUEST_SENT = "REQUEST_SENT",
    ACCEPTED = "ACCEPTED",
    REJECTED = "REJECT",
    REVOKE = "REVOKE"
}

export type ShareId = string
export interface SharesByPath {
    path: string,
    sharedBy: string,
    sharedByMe: boolean,
    shares: Share[]
}