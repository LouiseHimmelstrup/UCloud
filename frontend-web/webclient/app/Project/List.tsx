import {useCloudAPI} from "Authentication/DataHook";
import {emptyPage, ReduxObject} from "DefaultObjects";
import {MainContainer} from "MainContainer/MainContainer";
import * as Pagination from "Pagination";
import {listProjects, ListProjectsRequest, UserInProject} from "Project/index";
import * as React from "react";
import {connect} from "react-redux";
import {Dispatch} from "redux";
import {Page} from "Types";
import Button from "ui-components/Button";
import {Text, Icon} from "ui-components";
import Link from "ui-components/Link";
import VerticalButtonGroup from "ui-components/VerticalButtonGroup";
import {prettierString} from "UtilityFunctions";
import {List} from "ui-components";
import {updatePageTitle} from "Navigation/Redux/StatusActions";
import {setRefreshFunction} from "Navigation/Redux/HeaderActions";
import {ListRow} from "ui-components/List";
import {useHistory} from "react-router";
import {loadingAction} from "Loading";


// eslint-disable-next-line no-underscore-dangle
const _List: React.FunctionComponent<DispatchProps & {project?: string}> = props => {
    const [response, setFetchParams] = useCloudAPI<Page<UserInProject>, ListProjectsRequest>(
        listProjects({page: 0, itemsPerPage: 50}),
        emptyPage
    );

    React.useEffect(() => {
        props.setLoading(response.loading);
    }, [response.loading]);

    const history = useHistory();

    const reload = (): void =>
        setFetchParams(listProjects({page: response.data.pageNumber, itemsPerPage: response.data.itemsPerPage}));

    React.useEffect(() => {
        props.setPageTitle();
    }, []);

    React.useEffect(() => {
        props.setRefresh(reload);
        return () => props.setRefresh();
    }, [reload]);

    return (
        <MainContainer
            headerSize={58}
            header={<div />}
            main={(
                <Pagination.List
                    page={response.data}
                    pageRenderer={page => (
                        <List>
                            {page.items.map(e => {
                                const isSelected = e.projectId === props.project;
                                return (
                                    <ListRow
                                        key={e.projectId}
                                        navigate={() => history.push(`/projects/view/${encodeURIComponent(e.projectId)}`)}
                                        isSelected={false}
                                        select={() => undefined}
                                        left={<Text cursor="pointer">{e.title}</Text>}
                                        leftSub={<>
                                            <Text fontSize={0} pb="3px"><Icon mt="-2px" size="12px" name="id" /> {e.projectId}</Text>
                                            <Text fontSize={0} ml="4px"><Icon color="white" color2="black" mt="-2px" size="12px" name="user" />
                                                {" "}{prettierString(e.whoami.role)}
                                            </Text>
                                        </>}
                                        right={<>
                                            {isSelected ? <Link to="/projects/repositories"><Button mr="6px">Repositories</Button></Link> : null}
                                            {isSelected ? <Link to="/projects/groups/"><Button mr="38px">Groups</Button></Link> : null}
                                            {isSelected ? <Icon mr="44px" mt="9px" name="check" color="green" /> : (
                                                <Button onClick={() => props.setProject(e.projectId)}>Set active</Button>
                                            )}
                                        </>}
                                    />
                                );
                            })}
                        </List>
                    )}
                    loading={response.loading}
                    onPageChanged={newPage => setFetchParams(listProjects({page: newPage, itemsPerPage: 50}))}
                />
            )}
            sidebar={(
                <VerticalButtonGroup>
                    <Link to="/projects/create"><Button>Create</Button></Link>
                    <Button disabled={!props.project} color="red" onClick={() => props.setProject(undefined)}>Clear selection</Button>
                </VerticalButtonGroup>
            )}
        />
    );
};

interface ProjectSummaryProps {
    summary: UserInProject;
    isSelected: boolean;
    setProject(id: string): void;
}

interface DispatchProps {
    setProject: (id?: string) => void;
    setPageTitle: () => void;
    setRefresh: (refresh?: () => void) => void;
    setLoading: (loading: boolean) => void;
}

const mapStateToProps = (state: ReduxObject): {project?: string} => state.project;

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
    setPageTitle: () => dispatch(updatePageTitle("Projects")),
    setProject: id => dispatch({type: "SET_PROJECT", project: id}),
    setRefresh: refresh => dispatch(setRefreshFunction(refresh)),
    setLoading: loading => dispatch(loadingAction(loading))
});

export default connect(mapStateToProps, mapDispatchToProps)(_List);
