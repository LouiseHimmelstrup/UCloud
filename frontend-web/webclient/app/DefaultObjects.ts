import {SidebarOption, Page} from "Types";
import {Status} from "Navigation";
import {Analysis, DetailedApplicationSearchReduxState, RunsSortBy} from "Applications";
import {File, DetailedFileSearchReduxState} from "Files";
import {SortOrder, SortBy} from "Files";
import {DashboardStateProps} from "Dashboard";
import {Publication} from "Zenodo";
import {Notification} from "Notifications";
import {Upload} from "Uploader";
import {Activity, ActivityGroup, ActivityFilter} from "Activity";
import {Reducer} from "redux";
import {SimpleSearchStateProps} from "Search";
import * as ApplicationRedux from "Applications/Redux";
import * as AccountingRedux from "Accounting/Redux";
import * as SnackbarRedux from "Snackbar/Redux";
import * as FavoritesRedux from "Favorites/Redux";
import {defaultAvatar} from "UserSettings/Avataaar";
import {SidebarPages} from "ui-components/Sidebar";
import {ScrollResult} from "Scroll/Types";
import * as ProjectRedux from "Project/Redux";

export const DefaultStatus: Status = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended."
};

export enum KeyCode {
    ENTER = 13,
    ESC = 27
}

export const emptyPage: Page<any> = {items: [], itemsPerPage: 25, itemsInTotal: 0, pageNumber: 0, pagesInTotal: 0};

export enum AnalysesStatusMap {
    "PENDING",
    "IN PROGRESS",
    "COMPLETED"
}

export enum RightsNameMap {
    "NONE" = "None",
    "READ" = "Read",
    "READ_WRITE" = "Read/Write"
}

export enum SensitivityLevel {
    "INHERIT" = "Inherit",
    "PRIVATE" = "Private",
    "CONFIDENTIAL" = "Confidential",
    "SENSITIVE" = "Sensitive"
}

export type Sensitivity = keyof typeof SensitivityLevel;

export enum SensitivityLevelMap {
    INHERIT = "INHERIT",
    PRIVATE = "PRIVATE",
    CONFIDENTIAL = "CONFIDENTIAL",
    SENSITIVE = "SENSITIVE"
}

function getFilesSortingColumnOrDefault(columnIndex: 0 | 1): SortBy {
    const sortingColumn = window.localStorage.getItem(`filesSorting${columnIndex}`);
    if (sortingColumn && Object.values(SortBy).includes(sortingColumn)) return sortingColumn as SortBy;
    switch (columnIndex) {
        case 0:
            window.localStorage.setItem("filesSorting0", SortBy.MODIFIED_AT);
            return SortBy.MODIFIED_AT;
        case 1:
            window.localStorage.setItem("filesSorting1", SortBy.SIZE);
            return SortBy.SIZE;
    }
}

function getItemOrDefault<T, T2>(itemName: string, defaultValue: T, en: T2): T {
    const item = window.localStorage.getItem(itemName);
    if (item && Object.values(en).includes(item)) {
        return item as unknown as T;
    }
    return defaultValue;
}

export interface ComponentWithLoadingState {
    loading: boolean
    error?: string
}

export interface ComponentWithPage<T> extends ComponentWithLoadingState {
    page: Page<T>
}

export interface ComponentWithScroll<Item, OffsetType> extends ComponentWithLoadingState {
    scroll?: ScrollResult<Item, OffsetType>
}

export interface ResponsiveReduxObject {
    mediaType: string
    orientation: string
    lessThan: Record<string, boolean>
    greaterThan: Record<string, boolean>
    is: Record<string, boolean>
}

export const initResponsive = (): ResponsiveReduxObject => ({
    mediaType: "",
    orientation: "",
    lessThan: {},
    greaterThan: {},
    is: {},
});

export interface FilePreviewReduxState {
    file?: File
    error?: string
}

export interface FilesReduxObject extends ComponentWithPage<File> {
    sortOrder: SortOrder
    sortBy: SortBy
    path: string
    filesInfoPath: string
    fileSelectorError?: string
    sortingColumns: [SortBy, SortBy]
    fileSelectorLoading: boolean
    fileSelectorShown: boolean
    fileSelectorPage: Page<File>
    fileSelectorPath: string
    fileSelectorCallback: Function
    fileSelectorIsFavorites: boolean
    disallowedPaths: string[]
    invalidPath: boolean
}

export interface FileInfoReduxObject {
    file?: File
    error?: string
    activity: Page<Activity>
    loading: boolean
}

export interface AnalysisReduxObject extends ComponentWithPage<Analysis> {
    sortBy: RunsSortBy
    sortOrder: SortOrder
}

export interface NotificationsReduxObject {
    redirectTo: string
    items: Notification[]
    loading: boolean
    error?: string
}

export interface ZenodoReduxObject extends ComponentWithPage<Publication> {
    connected: boolean
}

export interface StatusReduxObject {
    status: Status
    title: string
    page: SidebarPages
    loading: boolean
}

export interface SidebarReduxObject {
    pp: boolean
    options: SidebarOption[]
    kcCount: number
}

export interface HeaderSearchReduxObject {
    prioritizedSearch: HeaderSearchType
    refresh?: () => void
}

export interface RunApplicationReduxObject {

}

export type ActivityReduxObject = ComponentWithScroll<ActivityGroup, number> & ActivityFilter;

export type HeaderSearchType = "files" | "applications" | "projects";

export interface UploaderReduxObject {
    uploads: Upload[]
    visible: boolean
    path: string
    allowMultiple: boolean
    onFilesUploaded: (p: string) => void
    error?: string
    loading: boolean
}

interface LegacyReducers {
    dashboard?: Reducer<DashboardStateProps>
    files?: Reducer<FilesReduxObject>
    uploader?: Reducer<UploaderReduxObject>
    status?: Reducer<StatusReduxObject>
    notifications?: Reducer<NotificationsReduxObject>
    analyses?: Reducer<AnalysisReduxObject>
    zenodo?: Reducer<ZenodoReduxObject>
    header?: Reducer<HeaderSearchReduxObject>
    sidebar?: Reducer<SidebarReduxObject>
    activity?: Reducer<ActivityReduxObject>
    detailedResult?: Reducer<DetailedResultReduxObject>
}

export type Reducers = LegacyReducers & ApplicationRedux.Reducers & AccountingRedux.Reducers;

export type DetailedResultReduxObject = ComponentWithPage<File>

export const initDetailedResult = (): DetailedResultReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined
});

/* FIXME */
interface LegacyReduxObject {
    dashboard: DashboardStateProps
    files: FilesReduxObject,
    uploader: UploaderReduxObject
    status: StatusReduxObject,
    notifications: NotificationsReduxObject
    analyses: AnalysisReduxObject
    zenodo: ZenodoReduxObject
    header: HeaderSearchReduxObject
    sidebar: SidebarReduxObject
    activity: ActivityReduxObject
    detailedResult: DetailedResultReduxObject
    simpleSearch: SimpleSearchStateProps
    detailedFileSearch: DetailedFileSearchReduxState
    detailedApplicationSearch: DetailedApplicationSearchReduxState
    fileInfo: FileInfoReduxObject
    avatar: AvatarReduxObject
    filePreview: FilePreviewReduxState
    responsive?: ResponsiveReduxObject
    project: ProjectRedux.State
    loading?: boolean
}

export type ReduxObject =
    LegacyReduxObject &
    ApplicationRedux.Objects &
    AccountingRedux.Objects &
    FavoritesRedux.Objects &
    SnackbarRedux.Wrapper;

export const initActivity = (): ActivityReduxObject => ({
    loading: false
});

export const initNotifications = (): NotificationsReduxObject => ({
    items: [],
    loading: false,
    redirectTo: "",
    error: undefined
});

export const initHeader = (): HeaderSearchReduxObject => ({
    prioritizedSearch: "files"
});

export const initStatus = (): StatusReduxObject => ({
    status: DefaultStatus,
    title: "",
    page: SidebarPages.None,
    loading: false
});

export const initDashboard = (): DashboardStateProps => ({
    favoriteFiles: [],
    recentFiles: [],
    recentAnalyses: [],
    notifications: [],
    favoriteLoading: false,
    recentLoading: false,
    analysesLoading: false
});

export function initObject(homeFolder: string): ReduxObject {
    return {
        dashboard: initDashboard(),
        files: initFiles(homeFolder),
        status: initStatus(),
        header: initHeader(),
        notifications: initNotifications(),
        analyses: initAnalyses(),
        zenodo: initZenodo(),
        sidebar: initSidebar(),
        uploader: initUploads(),
        activity: initActivity(),
        detailedResult: initDetailedResult(),
        simpleSearch: initSimpleSearch(),
        detailedApplicationSearch: initApplicationsAdvancedSearch(),
        detailedFileSearch: initFilesDetailedSearch(),
        fileInfo: initFileInfo(),
        avatar: initAvatar(),
        filePreview: initFilePreview(),
        project: ProjectRedux.initialState,
        ...ApplicationRedux.init(),
        ...AccountingRedux.init(),
        ...FavoritesRedux.init(),
        ...SnackbarRedux.init(),
        responsive: undefined,
    }
}


export const initFilePreview = () => ({
    file: undefined,
    error: undefined
});

export type AvatarReduxObject = typeof defaultAvatar & { error?: string };
export const initAvatar = (): AvatarReduxObject => ({...defaultAvatar, error: undefined});

export const initSimpleSearch = (): SimpleSearchStateProps => ({
    files: emptyPage,
    filesLoading: false,
    applications: emptyPage,
    applicationsLoading: false,
    errors: [],
    search: "",
    applicationSearch: initApplicationsAdvancedSearch(),
    fileSearch: initFilesDetailedSearch()
});

export const initAnalyses = (): AnalysisReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined,
    sortBy: RunsSortBy.createdAt,
    sortOrder: SortOrder.ASCENDING
});

export const initZenodo = (): ZenodoReduxObject => ({
    connected: false,
    loading: false,
    page: emptyPage
});

export const initSidebar = (): SidebarReduxObject => ({
    pp: false,
    kcCount: 0,
    options: []
});

export const initUploads = (): UploaderReduxObject => ({
    path: "",
    uploads: [],
    visible: false,
    allowMultiple: false,
    error: undefined,
    onFilesUploaded: () => null,
    loading: false
});

export const initFileInfo = (): FileInfoReduxObject => ({
    activity: emptyPage,
    loading: false
});

export const initFiles = (homeFolder: string): FilesReduxObject => ({
    page: emptyPage,
    sortOrder: getItemOrDefault("sortOrder", SortOrder.ASCENDING, SortOrder),
    sortBy: getItemOrDefault("sortBy", SortBy.PATH, SortBy),
    loading: false,
    error: undefined,
    path: "",
    filesInfoPath: "",
    sortingColumns: [getFilesSortingColumnOrDefault(0), getFilesSortingColumnOrDefault(1)],
    fileSelectorLoading: false,
    fileSelectorShown: false,
    fileSelectorPage: emptyPage,
    fileSelectorPath: homeFolder,
    fileSelectorIsFavorites: false,
    fileSelectorCallback: () => undefined,
    fileSelectorError: undefined,
    disallowedPaths: [],
    invalidPath: false
});

export const initFilesDetailedSearch = (): DetailedFileSearchReduxState => ({
    hidden: true,
    allowFolders: true,
    allowFiles: true,
    fileName: "",
    extensions: new Set(),
    tags: new Set(),
    sensitivities: new Set(),
    createdBefore: undefined,
    createdAfter: undefined,
    modifiedBefore: undefined,
    modifiedAfter: undefined,
    error: undefined,
    loading: false
});

export const initApplicationsAdvancedSearch = (): DetailedApplicationSearchReduxState => ({
    error: undefined,
    loading: false,
    hidden: true,
    appName: "",
    appVersion: "",
    tags: ""
});
