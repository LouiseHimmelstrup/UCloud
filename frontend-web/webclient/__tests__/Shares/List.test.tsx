import * as React from "react";
import List from "Shares/List";
import { create } from "react-test-renderer";
import { Provider } from "react-redux";
import { configureStore } from "Utilities/ReduxUtilities";
import { initFiles } from "DefaultObjects";
import files from "Files/Redux/FilesReducer";
import { MemoryRouter } from "react-router";
import { configure, shallow, mount } from "enzyme"
import * as Adapter from "enzyme-adapter-react-16";
import { shares } from "../mock/Shares";

configure({ adapter: new Adapter() });

describe("Shares List", () => {
    test.skip("Shares component", () => {
        expect(create(
            <Provider store={configureStore({ files: initFiles("/home/user@test.abc/") }, { files })}>
                <MemoryRouter>
                    <List />
                </MemoryRouter>
            </Provider >)).toMatchSnapshot();
    });

    test.skip("Shares component with shares", () => {
        let sharesListWrapper = shallow(
            <Provider store={configureStore({ files: initFiles("/home/user@test.abc/") }, { files })}>
                <MemoryRouter>
                    <List keepTitle={true} />
                </MemoryRouter>
            </Provider >);
        console.warn(shares.items);
        sharesListWrapper = sharesListWrapper.update();
        console.error(sharesListWrapper.find(List).dive().state());
        sharesListWrapper.find(List).dive().setState(() => ({ shares: shares.items }));
        console.error(sharesListWrapper.find(List).dive().state());
        expect(sharesListWrapper.html()).toMatchSnapshot();
    });
});