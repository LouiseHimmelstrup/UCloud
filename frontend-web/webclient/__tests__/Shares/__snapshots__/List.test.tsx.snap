// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shares List Shares component 1`] = `
.c10 {
  width: 32px;
  height: 32px;
  margin: 20px auto;
  -webkit-animation-name: jkjquP;
  animation-name: jkjquP;
}

.c10 > svg path:nth-child(1) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.04s;
  animation-delay: -0.04s;
}

.c10 > svg path:nth-child(2) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.08s;
  animation-delay: -0.08s;
}

.c10 > svg path:nth-child(3) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.12s;
  animation-delay: -0.12s;
}

.c10 > svg path:nth-child(4) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.16s;
  animation-delay: -0.16s;
}

.c10 > svg path:nth-child(5) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c10 > svg path:nth-child(6) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.24s;
  animation-delay: -0.24s;
}

.c10 > svg path:nth-child(7) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.28s;
  animation-delay: -0.28s;
}

.c10 > svg path:nth-child(8) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.32s;
  animation-delay: -0.32s;
}

.c10 > svg path:nth-child(9) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.36s;
  animation-delay: -0.36s;
}

.c10 > svg path:nth-child(10) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.4s;
  animation-delay: -0.4s;
}

.c10 > svg path:nth-child(11) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.44s;
  animation-delay: -0.44s;
}

.c10 > svg path:nth-child(12) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.48s;
  animation-delay: -0.48s;
}

.c10 > svg path:nth-child(13) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.52s;
  animation-delay: -0.52s;
}

.c10 > svg path:nth-child(14) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.56s;
  animation-delay: -0.56s;
}

.c10 > svg path:nth-child(15) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.6s;
  animation-delay: -0.6s;
}

.c10 > svg path:nth-child(16) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.64s;
  animation-delay: -0.64s;
}

.c10 > svg path:nth-child(17) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.68s;
  animation-delay: -0.68s;
}

.c10 > svg path:nth-child(18) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.72s;
  animation-delay: -0.72s;
}

.c0 {
  margin-left: 190px;
  padding-top: 48px;
  padding-bottom: 14px;
  padding-left: 14px;
  padding-right: 0;
  background-color: #fff;
}

.c1 {
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 204px;
  padding-right: 14px;
  width: 100%;
  height: 55px;
  background-color: #fff;
  top: 48px;
}

.c8 {
  padding-top: 55px;
  padding-right: 14px;
}

.c2 {
  position: absolute;
  top: 48px;
  left: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  cursor: pointer;
  font-size: 20px;
  margin-right: 1em;
}

.c5 {
  border-bottom: 2px solid #c9d3df;
  cursor: pointer;
}

.c7 {
  border-bottom: 3px solid #006aff;
}

.c3 {
  position: fixed;
  z-index: 50;
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c9 {
    display: none;
  }
}

@media screen and (min-width:1024px) and (max-width:1279px) {
  .c9 {
    display: none;
  }
}

@media screen and (min-width:1280px) {
  .c9 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2 c3"
    height={55}
    width={1}
  >
    <div
      className="c4 c5"
    >
      <div
        className="c6 c7"
        cursor="pointer"
        fontSize={3}
        onClick={[Function]}
        selected={true}
      >
        Shared with Me
      </div>
      <div
        className="c6 "
        cursor="pointer"
        fontSize={3}
        onClick={[Function]}
        selected={false}
      >
        Shared by Me
      </div>
    </div>
  </div>
  <div
    className="c8"
  >
    <div
      className="c9"
    />
    <div
      className="c10"
      data-tag="loading-spinner"
      height={32}
      width={32}
    >
      <svg
        height={32}
        id="svg"
        version="1.1"
        viewBox="0 0 220 220"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M80.296,0.12l14.673,54.761l40.088,-40.088l-54.761,-14.673Z"
        />
        <path
          d="M149.73,69.554l-14.673,-54.761l-40.088,40.088l54.761,14.673Z"
        />
        <path
          d="M189.818,29.466l-54.761,-14.673l14.673,54.761l40.088,-40.088Z"
        />
        <path
          d="M204.491,84.228l-14.673,-54.762l-40.088,40.088l54.761,14.674Z"
        />
        <path
          d="M164.403,124.316l40.088,-40.088l-54.761,-14.674l14.673,54.762Z"
        />
        <path
          d="M219.165,138.989l-14.674,-54.761l-40.088,40.088l54.762,14.673Z"
        />
        <path
          d="M179.077,179.077l-14.674,-54.761l54.762,14.673l-40.088,40.088Z"
        />
        <path
          d="M124.316,164.403l54.761,14.674l-14.674,-54.761l-40.087,40.087Z"
        />
        <path
          d="M138.989,219.165l40.088,-40.088l-54.761,-14.674l14.673,54.762Z"
        />
        <path
          d="M84.228,204.491l54.761,14.674l-14.673,-54.762l-40.088,40.088Z"
        />
        <path
          d="M69.554,149.73l14.674,54.761l40.088,-40.088l-54.762,-14.673Z"
        />
        <path
          d="M29.466,189.818l54.762,14.673l-14.674,-54.761l-40.088,40.088Z"
        />
        <path
          d="M14.793,135.057l14.673,54.761l40.088,-40.088l-54.761,-14.673Z"
        />
        <path
          d="M54.881,94.969l-40.088,40.088l54.761,14.673l-14.673,-54.761Z"
        />
        <path
          d="M0.12,80.296l14.673,54.761l40.088,-40.088l-54.761,-14.673Z"
        />
        <path
          d="M40.208,40.208l-40.088,40.088l54.761,14.673l-14.673,-54.761Z"
        />
        <path
          d="M94.969,54.881l-54.761,-14.673l14.673,54.761l40.088,-40.088Z"
        />
        <path
          d="M80.296,0.12l-40.088,40.088l54.761,14.673l-14.673,-54.761Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
